tests/test_repository.py

```diff
@@ -3,315 +3,353 @@
-    assert repo.svn_info_show_item("some-mock-item") == mock_subprocess.stdout
-    assert (
-        f"svn info --show-item some-mock-item {MOCK_CWD}/src/trunk"
-        in mock_subprocess.commands
-    )
 
-    # Success case: test leading and trailing white space is removed from standard output
-    mock_subprocess = MockSubprocessWrapper()
-    mock_subprocess.stdout = " \n\n mock standard output \n\n"
-    repo = get_mock_repo(mock_subprocess)
-    assert repo.svn_info_show_item("some-mock-item") == mock_subprocess.stdout.strip()
-    assert (
-        f"svn info --show-item some-mock-item {MOCK_CWD}/src/trunk"
-        in mock_subprocess.commands
-    )
+    def test_svn_info_command_execution(self, repo, mock_subprocess_handler, mock_cwd):
+        """Success case: call svn info command and get result."""
+        assert (
+            repo.svn_info_show_item("some-mock-item") == mock_subprocess_handler.stdout
+        )
+        assert (
+            f"svn info --show-item some-mock-item {mock_cwd}/src/trunk"
+            in mock_subprocess_handler.commands
+        )
+
+    def test_white_space_removed_from_standard_output(
+        self, repo, mock_subprocess_handler
+    ):
+        """Success case: test leading and trailing white space is removed from standard output."""
+        mock_subprocess_handler.stdout = " \n\n mock standard output \n\n"
+        assert (
+            repo.svn_info_show_item("some-mock-item")
+            == mock_subprocess_handler.stdout.strip()
+        )
 
 
-def test_pre_build():
+class TestPreBuild:
     """Tests for `CableRepository.pre_build()`."""
-    repo_dir = MOCK_CWD / internal.SRC_DIR / "trunk"
-    offline_dir = repo_dir / "offline"
-    offline_dir.mkdir(parents=True)
-    (offline_dir / "Makefile").touch()
-    (offline_dir / "parallel_cable").touch()
-    (offline_dir / "serial_cable").touch()
-    (offline_dir / "foo.f90").touch()
-
-    # Success case: test source files and scripts are copied to .tmp
-    repo = get_mock_repo()
-    repo.pre_build()
-    assert (offline_dir / ".tmp" / "Makefile").exists()
-    assert (offline_dir / ".tmp" / "parallel_cable").exists()
-    assert (offline_dir / ".tmp" / "serial_cable").exists()
-    assert (offline_dir / ".tmp" / "foo.f90").exists()
-    shutil.rmtree(offline_dir / ".tmp")
-
-    # Success case: test non-verbose standard output
-    repo = get_mock_repo()
-    with contextlib.redirect_stdout(io.StringIO()) as buf:
+
+    @pytest.fixture(autouse=True)
+    def _setup(self, repo):
+        """Setup precondition for `CableRepository.pre_build()`."""
+        (internal.SRC_DIR / repo.name / "offline").mkdir(parents=True)
+        (internal.SRC_DIR / repo.name / "offline" / "Makefile").touch()
+        (internal.SRC_DIR / repo.name / "offline" / "parallel_cable").touch()
+        (internal.SRC_DIR / repo.name / "offline" / "serial_cable").touch()
+        (internal.SRC_DIR / repo.name / "offline" / "foo.f90").touch()
```

Comment:
Should we add a teardown?

Reply:
I figured all directories under the `mock_cwd` are removed at the end by the `_run_around_tests` fixture which is why I haven't been adding a teardown step.

Reply:
Could we have a comment at the file level to remind us of that? Happy to have the same comment in all `test_*` files. But I am sure to have forgotten in a few days otherwise.

Reply:
Sure, we can put a comment the module docstring, e.g:
```python
"""`pytest` tests for `repository.py`.

Note: explicit teardown for generated files and directories are not required as
the working directory used for testing is cleaned up in the `_run_around_tests`
pytest autouse fixture.
"""
```